### PR TITLE
Homework1

### PR DESCRIPTION
**Pull request by cecelia-tech**
Submitting homework 1 tasks 1-3.

### CODE REVIEW

File: Homework1/ISBN/Isbn.cs [hunk 1]
```diff
@@ -0,0 +1,84 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ISBN
+{
+    class Isbn
```
Comment (radislavkireev) [inline Homework1/ISBN/Isbn.cs]: add protection modifier explicitly: "internal"

File: Homework1/ISBN/Isbn.cs [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+            {
+                int digit = 11 - remainder;
+
+                return digit.ToString();
+            }
+        }
+
+
+
+        public void Print(string isbnCode)
```
Comment (radislavkireev) [inline Homework1/ISBN/Isbn.cs]: This method is not required, drop it

File: Homework1/ISBN/Isbn.cs [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ISBN
+{
+    class Isbn
+    {
+        //Method combines ISBN code
+        public void IsbnNumericCode(string userInput)
```
Comment (radislavkireev) [inline Homework1/ISBN/Isbn.cs]: The methods names does not correspond to what they actually do. 
1 . Method name should start with verb what describes the action it does. 
2. Method code should do what it name declares. So, if it is called GetSomething it should return some value, but not print it or save for example. 

Lets update the class next way:
1.public void IsbnNumericCode(string userInput) -> rename to public string  GetIsbnNumericCode(string userInput). The method should NOT print numeric code, leave it for Program.cs, it should only return ISBN number. 

2. public string StringToIntArrayConvertor(string numbersInString) -> rename to public int[] ConvertStringToIntArray(string numbersInString). The method should NOT call MultiplicationOfNumbers, it should return 
arrayOfNum.

3. public string MultiplicationOfNumbers(int[] arrayOfNumbers) => rename to public int SumAndMultiplyNumbers(int[] arrayOfNumbers) . The method should NOT call CheckDigit, but return sumOfMultipliedNumbers. 

4. Update GetIsbnNumericCode, so it calls methods in the following order: 
```
{
    string finalIsbn = "ISBN ";
    //....
    var arrayOfNum= ConvertStringToIntArray(userInput);
    int sumOfMultipliedNumbers= SumAndMultiplyNumbers(arrayOfNum);
   string checkDigit = CheckDigit(sumOfMultipliedNumbers);
  
    return finalIsbn + checkDigit;
 
}
```

File: Homework1/SumOfArray/SumBetweenNumbers.cs [hunk 4]
```diff
@@ -0,0 +1,83 @@
+﻿using System;
+
+namespace SumOfArray
+{
+    class SumBetweenNumbers
+    {
+        //Calculates the sum of numbers between the smallest and the biggest values in the array
+        public void SumOfRequaredNumbers(string userInputNumbers)
```
Comment (radislavkireev) [inline Homework1/SumOfArray/SumBetweenNumbers.cs]: rename to PrintSumOfRequiredNumbers

File: Homework1/SumOfArray/SumBetweenNumbers.cs [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+            {
+                if (userInputNumbers[minValueIndex] > userInputNumbers[i])
+                {
+                    minValueIndex = i;
+                }
+            }
+            return minValueIndex;
+        }
+
+        public void Print(int sumOfPartOfArray)
```
Comment (radislavkireev) [inline Homework1/SumOfArray/SumBetweenNumbers.cs]: This method is not required, just call Console.WriteLine at the caller.

File: Homework1/SumOfArray/SumBetweenNumbers.cs [hunk 6]
```diff
@@ -0,0 +1,83 @@
+﻿using System;
+
+namespace SumOfArray
+{
+    class SumBetweenNumbers
```
Comment (radislavkireev) [inline Homework1/SumOfArray/SumBetweenNumbers.cs]: add internal

File: Homework1/SumOfArray/SumBetweenNumbers.cs [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+                {
+                    sumOfRequaredNumbers += convertedNumbers[i];
+                }
+            }
+
+            Print(sumOfRequaredNumbers);
+        }
+
+        //Converts user input string to int array
+        public int[] StringToInt(string userInpput)
```
Comment (radislavkireev) [inline Homework1/SumOfArray/SumBetweenNumbers.cs]: rename method, so it starts with a verb

File: Homework1/SumOfArray/SumBetweenNumbers.cs [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+            for (int i = 0; i < userInputNumbers.Length; i++)
+            {
+                userInputNumbers[i] = int.Parse(arrayOfStrings[i]);
+            }
+
+            return userInputNumbers;
+        }
+
+        //Calculates the biggest most right-hand side number in the array
+        public int MaxValueIndex(int[] userInputNumbers)
```
Comment (radislavkireev) [inline Homework1/SumOfArray/SumBetweenNumbers.cs]: rename method, so it starts with a verb

File: Homework1/SumOfArray/SumBetweenNumbers.cs [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+                if (userInputNumbers[maxValueIndex] <= userInputNumbers[i])
+                {
+                    maxValueIndex = i;
+                }
+            }
+            return maxValueIndex;
+        }
+
+        //Calculates the smallest most left-hand side number in the array
+        public int MinValueIndex(int[] userInputNumbers)
```
Comment (radislavkireev) [inline Homework1/SumOfArray/SumBetweenNumbers.cs]: rename method, so it starts with a verb

File: Homework1/Ternary/TernaryConvertor.cs [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+            else
+            {
+                //ternary number system first number always is 1
+                ternaryString.Insert(0, 1);
+            }
+
+            return ternaryString;
+        }
+
+        public void NumberPrint(int number)
```
Comment (radislavkireev) [inline Homework1/Ternary/TernaryConvertor.cs]: Drop this method, just call Console.WriteLine

File: Homework1/Ternary/TernaryConvertor.cs [hunk 11]
```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using System.Text;
+
+namespace Ternary
+{
+    class TernaryConvertor
+    {
+        //Method checks for numbers satisfying the task in the range given
+        public void CheckForAppropriateInteger(int numberFrom, int numberTo)
```
Comment (radislavkireev) [inline Homework1/Ternary/TernaryConvertor.cs]: Rename to PrintNumberWith2Twos

File: Homework1/Ternary/TernaryConvertor.cs [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+                {
+                    numberOf2InString++;
+                }
+            }
+            return numberOf2InString == 2;
+        }
+
+        //This method converts given number  to ternary number system and returns
+        //it as a string
+        public StringBuilder ConvertToTernary(int number)
```
Comment (radislavkireev) [inline Homework1/Ternary/TernaryConvertor.cs:43]: This method does not work correctly. 
The alg could be viewed at https://www.geeksforgeeks.org/ternary-number-system-or-base-3-numbers/ as an example. 

Here is the output for range 0..30.
As you can see it starts from 10, when should from 01. 
Also, it repeats ternary representations twice. 

10
11
12
10
11
12
10
11
12
100
101
102
110
111
112
120
121
122
100
101
102
110
111
112
120
121
122
1000
1001
1002
1010

**Review by radislavkireev — CHANGES_REQUESTED** [review]

**Review by rkireevLT — APPROVED** [review]

**Review by radislavkireev — APPROVED** [review]

**Pull request merged**